### PR TITLE
6371 migrate sharding tags

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**

https://github.com/gravitee-io/gravitee-api-management/pull/245

**Description**

next step for migration of sharding tags

**Additional context**
image

### CODE REVIEW

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-ywhwagfdqg.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

File: gravitee-apim-console-webui/src/shared/components/gio-clipboard/gio-clipboard.base.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+export abstract class GioClipboardComponent {
+  tooltipMessage = 'Copy to clipboard';
+  tooltipHideDelay = 0;
+  clicked = false;
+
+  @Input() contentToCopy: string;
+
+  onCopied(success: boolean, tooltip: MatTooltip) {
+    tooltip.message = success ? 'Copied !' : 'Failed to copy !';
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/shared/components/gio-clipboard/gio-clipboard.base.component.ts]:
```suggestion
    tooltip.message = success ? 'Copied!' : 'Failed to copy!';
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 2]
```diff
@@ -148,8 +154,53 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   onEditTagClicked() {}
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  onDeleteTagClicked() {}
+  onDeleteTagClicked(tag: TagTableDS[number]) {
+    const entrypointsToUpdate = this.entrypoints.filter((entrypoint) => entrypoint.tags.includes(tag.id));
+    let entrypointsInfoMessage = '';
+    if (entrypointsToUpdate.length === 1) {
+      entrypointsInfoMessage = `<br>⚠️ The tag will be removed for the entrypoint <strong>${entrypointsToUpdate[0].value}</strong>.`;
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
      entrypointsInfoMessage = `<br>The tag will be removed for the entrypoint <strong>${entrypointsToUpdate[0].value}</strong>.`;
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 3]
```diff
@@ -148,8 +154,53 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
+    } else if (entrypointsToUpdate.length > 1) {
+      entrypointsInfoMessage = `
+        <br>⚠️ The tag will be removed from all these entrypoints:
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
        <br>The tag will be removed from all these entrypoints:
```

File: gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts [hunk 4]
```diff
@@ -148,8 +154,53 @@ export class OrgSettingsTagsComponent implements OnInit, OnDestroy {
...
+        <ul>
+          <li><strong>${entrypointsToUpdate.map((e) => e.value).join('</strong></li><li><strong>')}</li>
+        </ul>`;
+    }
+
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData>(GioConfirmDialogComponent, {
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/tags/org-settings-tags.component.ts]:
```suggestion
      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
```

**Review by gaetanmaisse — APPROVED** [review]

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

📚 View the storybook of this branch [here](https://612657caa8e859003a8a6430-hflzsauddm.chromatic.com)

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6371-migrate-sharding-tags/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Pull request merged**